component/common/src/main/java/org/wso2/carbon/extension/identity/emailotp/common/util/Utils.java

```diff
@@ -0,0 +1,190 @@
+        boolean triggerNotification = Boolean.parseBoolean(StringUtils.trim(
+                properties.getProperty(Constants.EMAIL_OTP_TRIGGER_OTP_NOTIFICATION)));
+        configs.setTriggerNotification(triggerNotification);
+
+        boolean showFailureReason = Boolean.parseBoolean(StringUtils.trim(
+                properties.getProperty(Constants.EMAIL_OTP_SHOW_FAILURE_REASON)));
+        configs.setShowFailureReason(showFailureReason);
+
+        boolean isAlphaNumericOtp = Boolean.parseBoolean(StringUtils.trim(
+                properties.getProperty(Constants.EMAIL_OTP_ALPHA_NUMERIC_OTP)));
+        configs.setAlphaNumericOTP(isAlphaNumericOtp);
+
+        String otpLengthValue = StringUtils.trim(properties.getProperty(
+                Constants.EMAIL_OTP_LENGTH));
+        int otpLength = StringUtils.isNumeric(otpLengthValue) ?
+                Integer.parseInt(otpLengthValue) : Constants.DEFAULT_OTP_LENGTH;
+        configs.setOtpLength(otpLength);
+
+        String otpValidityPeriodValue =
+                StringUtils.trim(properties.getProperty(Constants.EMAIL_OTP_VALIDITY_PERIOD));
+        int otpValidityPeriod = StringUtils.isNumeric(otpValidityPeriodValue) ?
+                Integer.parseInt(otpValidityPeriodValue) * 1000 : Constants.DEFAULT_EMAIL_OTP_EXPIRY_TIME;
+        configs.setOtpValidityPeriod(otpValidityPeriod);
+
+        // If not defined, defaults to 'zero' to renew always.
+        String otpRenewIntervalValue = StringUtils.trim(
+                properties.getProperty(Constants.EMAIL_OTP_RENEWAL_INTERVAL));
+        int otpRenewalInterval = StringUtils.isNumeric(otpRenewIntervalValue) ?
+                Integer.parseInt(otpRenewIntervalValue) * 1000 : 0;
+        configs.setOtpRenewalInterval(otpRenewalInterval);
+
+        if (otpRenewalInterval >= otpValidityPeriod) {
+            throw Utils.handleServerException(Constants.ErrorMessage.SERVER_INVALID_RENEWAL_INTERVAL_ERROR,
+                    String.valueOf(otpRenewalInterval));
+        }
+
+        String otpResendThrottleIntervalValue = StringUtils.trim(
+                properties.getProperty(Constants.EMAIL_OTP_RESEND_THROTTLE_INTERVAL));
+        int resendThrottleInterval = StringUtils.isNumeric(otpResendThrottleIntervalValue) ?
+                Integer.parseInt(otpResendThrottleIntervalValue) * 1000 :
+                Constants.DEFAULT_EMAIL_RESEND_THROTTLE_INTERVAL;
+        configs.setResendThrottleInterval(resendThrottleInterval);
+
+        // Should we send the same OTP upon the next generation request? Defaults to 'false'.
+        boolean resendSameOtp = (otpRenewalInterval > 0) && (otpRenewalInterval < otpValidityPeriod);
+        configs.setResendSameOtp(resendSameOtp);
+
+        // Defaults to 'true' with an interval of 30 seconds.
+        boolean resendThrottlingEnabled = resendThrottleInterval > 0;
+        configs.setResendThrottlingEnabled(resendThrottlingEnabled);
+    }
+
+    public static String getHash(String text) {
```

Comment:
Better to add method comments to the public methods

Reply:
Addressed in 148a8f957bdd2c4d574fb53e81284dfc017db263